frontend/hub/api/utils.tsx

```diff
@@ -7,11 +7,49 @@ import {
   postHubRequest,
   putHubRequest,
 } from './request';
+import { useTranslation } from 'react-i18next';
+import { requestGet } from '../../common/crud/Data';
 
 function getBaseAPIPath() {
   return process.env.HUB_API_PREFIX;
 }
 
+export function useRepositoryBasePath(name: string, pulp_href?: string): Promise<unknown> {
+  const { t } = useTranslation();
+  return Promise.all([
+    pulp_href
+      ? Promise.resolve({ name, pulp_href })
+      : requestGet<{ data: { results: unknown } }>(
+          pulpAPI`/repositories/ansible/ansible/?name=${name}&limit=1`
+        ).then(firstResult),
+    requestGet<{ data: { results: unknown } }>(
+      pulpAPI`/distributions/ansible/ansible/?name=${name}&limit=1`
+    ).then(firstResult),
+  ]).then(async ([repository, distribution]) => {
+    if (!repository) {
+      return Promise.reject(t`Failed to find repository ${name}`);
```

Comment:
For what I have seen in the code basis, we are adding the standard like like below.
```suggestion
      return Promise.reject(new Error(t`Failed to find repository ${name}`));
```

Reply:
Hey Kersom, 
I've added some console logs in CollectionDetails.tsx that use `useRepositoryBasePath()` function. The issue (I think) is that the Promise returns the object not the result, and I wanted ask where it should be resolved. 

The console logs in the download() function can be ignored for now, just stuck them in there to confirm values. 

Thank you~

Reply:
Hey @ShaiahWren the useRepositoryBasePath is returning a Promised that was not yet resolved yet.

With the current implementation we could do something like


```js
useRepositoryBasePath(name, pulp_href).then(response => console.log(response))

```

We could either update `useRepositoryBasePath` to return the resolved promised, for instance adding a `useEffect` and  `useState` to that hook. We could also add, error, and isLoading for example as part of the response.

Or on the component consuming useRepositoryBasePath add an `useEffect` and `useState` to fetch the data once the component mounts then fetch data, and update the state.

There are other hooks fetching data that they are adding an inner async function using useGetRequest as well, take a look at useRepoQueryOptions for example.